src/test/java/harness/utils/ExternalBQDatasets.java

```diff
@@ -21,44 +29,95 @@ public static String randomDatasetId() {
   }
 
   /**
-   * Grant a given user or group reader access to a dataset. This method uses SA credentials to set
-   * IAM policy on a bucket in an external (to WSM) project.
+   * Delete a dataset in the external project. This method uses SA credentials that have permissions
+   * on the external (to WSM) project.
    */
-  public static void grantReadAccess(Dataset dataset, Acl.Entity userOrGroup) throws IOException {
-    BigQuery bigQuery = getBQClient();
-    ArrayList<Acl> acls = new ArrayList<>(dataset.getAcl());
-    acls.add(Acl.of(userOrGroup, Acl.Role.READER));
-    bigQuery.update(dataset.toBuilder().setAcl(acls).build());
+  public static void deleteDataset(DatasetReference datasetRef)
+      throws IOException, GeneralSecurityException {
+    getBQClientWrapper()
+        .datasets()
+        .delete(datasetRef.getProjectId(), datasetRef.getDatasetId())
+        .setDeleteContents(true)
+        .execute();
   }
 
-  /** Helper method to build the BQ client object with SA credentials. */
-  public static BigQuery getBQClient() throws IOException {
-    return getBQClient(TestExternalResources.getSACredentials());
+  /**
+   * Helper enum to distinguish between two different types of IAM members: users and groups. This
+   * information could be conveyed using a boolean flag, but it makes the method signature more
+   * readable to use an enum instead.
+   */
+  public enum IamMemberType {
+    USER,
+    GROUP;
+  }
+
+  /**
+   * Grant a given user or group reader access to a dataset. This method uses SA credentials that
+   * have permissions on the external (to WSM) project.
+   */
+  public static void grantReadAccess(
+      DatasetReference datasetRef, String memberEmail, IamMemberType memberType)
+      throws IOException, GeneralSecurityException {
+    BigQueryCow bigQuery = getBQClientWrapper();
+    Dataset datasetToUpdate =
+        bigQuery.datasets().get(datasetRef.getProjectId(), datasetRef.getDatasetId()).execute();
+    List<Dataset.Access> accessToUpdate = datasetToUpdate.getAccess();
+
+    Dataset.Access newAccess = new Dataset.Access().setRole("READER");
+    if (memberType.equals(IamMemberType.USER)) {
+      newAccess.setUserByEmail(memberEmail);
+    } else {
+      newAccess.setGroupByEmail(memberEmail);
+    }
+    accessToUpdate.add(newAccess);
+    datasetToUpdate.setAccess(accessToUpdate);
+
+    bigQuery
+        .datasets()
+        .update(datasetRef.getProjectId(), datasetRef.getDatasetId(), datasetToUpdate)
+        .execute();
   }
 
-  /** Helper method to build the BQ client object with the given credentials. */
-  public static BigQuery getBQClient(GoogleCredentials credentials) throws IOException {
+  /**
+   * Helper method to build the CRL wrapper around the BQ client object with SA credentials that
+   * have permissions on the external (to WSM) project.
+   */
+  private static BigQueryCow getBQClientWrapper() throws IOException, GeneralSecurityException {
```

Comment:
nit: consider keeping terminology consistent with CRL `getBigQueryCow` or `getBQCow`.

Reply:
changed to `getBQCow`